mmdet/structures/bbox/horizontal_bbox.py

```diff
@@ -0,0 +1,324 @@
+            bboxes * scale_factor
+
+    def resize_(self, scale_factor: Tuple[float, float]) -> None:
+        """Inplace resize the box width and height w.r.t scale_factor.
+
+        Note:
+            Both ``rescale_`` and ``resize_`` will enlarge or shrink bboxes
+            w.r.t ``scale_facotr``. The difference is that ``resize_`` only
+            changes the width and the height of bboxes, but ``rescale_`` also
+            rescales the box centers simultaneously.
+
+        Args:
+            scale_factor (Tuple[float, float]): factors for scaling box
+                shapes. The length should be 2.
+        """
+        bboxes = self.tensor
+        assert len(scale_factor) == 2
+        ctrs = (bboxes[..., 2:] + bboxes[..., :2]) / 2
+        wh = bboxes[..., 2:] - bboxes[..., :2]
+        scale_factor = bboxes.new_tensor(scale_factor)
+        wh = wh * scale_factor
+        xy1 = ctrs - 0.5 * wh
+        xy2 = ctrs + 0.5 * wh
+        self.tensor = torch.cat([xy1, xy2], dim=-1)
+
+    def is_bboxes_inside(self, img_shape: Tuple[int, int]) -> BoolTensor:
+        """Find bboxes inside the image.
+
+        In ``HorizontalBoxes``, as long as a part of the box is inside the
+        image, this box will be regarded as True.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+
+        Returns:
+            BoolTensor: Index of the remaining bboxes. Assuming the original
+            horizontal boxes have shape (m, n, 4), the output has shape (m, n).
+        """
+        img_h, img_w = img_shape
+        bboxes = self.tensor
+        return (bboxes[..., 0] < img_w) & (bboxes[..., 2] > 0) \
+            & (bboxes[..., 1] < img_h) & (bboxes[..., 3] > 0)
+
+    def find_inside_points(self,
+                           points: Tensor,
+                           is_aligned: bool = False) -> BoolTensor:
+        """Find inside box points. Bboxes dimension must be 2.
+
+        Args:
+            points (Tensor): Points coordinates. Has shape of (m, 2).
+            is_aligned (bool): Whether ``points`` has been aligned with bboxes
+                or not. If True, the length of bboxes and ``points`` should be
+                the same. Defaults to False.
+
+        Returns:
+            BoolTensor: Index of inside box points. Assuming the boxes has
+            shape of (n, 4), if ``is_aligned`` is False. The index has
+            shape of (m, n). If ``is_aligned`` is True, m should be equal to n
+            and the index has shape of (m, ).
+        """
+        bboxes = self.tensor
+        assert bboxes.dim() == 2, 'bboxes dimension must be 2.'
+
+        if not is_aligned:
+            bboxes = bboxes[None, :, :]
+            points = points[:, None, :]
+        else:
+            assert bboxes.size(0) == points.size(0)
+
+        x_min, y_min, x_max, y_max = bboxes.unbind(dim=-1)
+        return (points[..., 0] > x_min) & (points[..., 0] < x_max) & \
```

Comment:
`points[..., 0] > x_min` or `points[..., 0] >= x_min`?

Reply:
Good suggestion! I prefer to use `points[..., 0] >= x_min`.